Straume/Zeptoparsec.lean

```diff
@@ -1 +1,187 @@
+import Straume.Iterator
+open Iterator
+
+/-
+
+To make Zeptoparsec work on another type γ, define all the necessary
+scaffolding in Iterator.lean, currently:
+- instance Iterable γ μ, where μ is the type that γ semantically consists of
+
+-/
+
+/- TODO: somehow make most of σ's implicit -/
+
 namespace Zeptoparsec
+
+variable (σ : Type) [Iterable σ ε] [Inhabited σ] [DecidableEq σ] [DecidableEq ε] [Inhabited ε]
```

Comment:
@arthurpaulino how do you think we can avoid an API which always requires reminding all the underlying functions of Sigma type.

cc @ixahedron can you please paste your `anyChar` snippet here for more context to @arthurpaulino ?

Reply:
@cognivore I'm not sure I understand your question, but I suspect you'd want to use `section Foo ... end Foo` to isolate arguments declared with the `variable` keyword

Reply:
Well, the issue is that this variable is used everywhere, polluting the API.

Reply:
Do I understand correctly that the issue is that for some functions you don't need all those instances?

If so, you can compartmentalize them with different variables besides `σ`

Reply:
> Yes, that's correct!

That's a wonderful property, thank you for discovering it. I think that all the Lean programmers should know about it.